doc: improve example for Error.captureStackTrace() (#46886)
nodejs/node: doc/api/errors.md

```js
function MyError() {

>   // To avoid calculating the stack trace twice, the stack trace limit is
>   // to zero. Before that, the current stack trace limit is saved so it can be
>   // restored later.

Reviewer: ```suggestion
  // Set the stack trace limit to zero to avoid calculating the stack trace twice.
```

```js
function MyError() {

>   const instance = new Error()

Reviewer: ```suggestion
  const { stackTraceLimit } = Error
  Error.stackTraceLimit = 0
  const instance = new Error()
  Error.stackTraceLimit = stackTraceLimit
```

Without this, the stack frames would be computed twice. That would be a lot of overhead that is not needed. Creating errors is actually quite expensive CPU wise.

Author: Good to know! I'll add a comment that explains this because otherwise I think it's a little confusing.

new MyError().stack;
```

> We recommend using classes that extend `Error` for custom errors:

Reviewer: Error.captureStackTrace may even be used in combination with custom errors. It's usage is mainly to avoid specific stack frames. It's therefore not really an alternative to classes.

Author: Oh right, that makes sense. I've only thought about hiding constructors, which is not necessary when you subclass `Error`. I'll just get rid of the whole 'We recommend...' part including the code example.
